security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -508,10 +515,9 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 		}
 
 		retry++
-		backOffInMilliSec = retry * backOffInMilliSec
-		randomTime := rand.Intn(initialBackOffIntervalInMilliSec)
-		time.Sleep(time.Duration(backOffInMilliSec+randomTime) * time.Millisecond)
-		log.Warnf("Failed to sign cert for %q: %v, will retry in %d millisec", resourceName, err, backOffInMilliSec)
+		backOffInMilliSec = rand.Int63n(retry * initialBackOffIntervalInMilliSec)
+		time.Sleep(time.Duration(backOffInMilliSec) * time.Millisecond)
```

Comment:
same question here - what are the implications of sleeping synchronously here?

Reply:
Same as above.